### PR TITLE
4 intial radio drive

### PR DESCRIPTION
**Pull request by David-Rey**
Radio Driver

### CODE REVIEW

File: EosPayload/drivers/RadioDriver.py [hunk 1]
```diff
@@ -1,7 +1,91 @@
...
+from datetime import datetime
+from queue import PriorityQueue
 import logging
+import time
+
+from digi.xbee.devices import XBeeDevice
+from digi.xbee.devices import RemoteXBeeDevice
+from digi.xbee.devices import XBee64BitAddress
+
+import EosLib.packet.data_header
```
Comment (thomasmholder) [inline EosPayload/drivers/RadioDriver.py:10]: These imports are unused

File: EosPayload/drivers/RadioDriver.py [hunk 2]
```diff
@@ -1,7 +1,91 @@
...
+        return Device.RADIO
+
+    @staticmethod
+    def get_device_name() -> str:
+        return "radio-driver"
+
     def device_read(self, logger: logging.Logger) -> None:
-        return 0
+        # Receives data from radio and sends it to MQTT
+        def data_receive_callback(xbee_message):
```
Comment (thomasmholder) [inline EosPayload/drivers/RadioDriver.py:54]: Is there a reason these are defined inside of device_read?

File: EosPayload/lib/mqtt/client.py.orig [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    def register_subscriber(self, topic: Topic, callback: Callable) -> None:
+        """ Receive an MQTT message.
+            Starts listening for messages of the Topic and calling Callback on them.
+            Async (Non-Blocking).
+
+        :param topic: the topic to filter for
+        :param callback: a function taking 3 parameters: (client, userdata, message)
+        """
+        self.message_callback_add(topic, callback)
+<<<<<<< HEAD
```
Comment (thomasmholder) [inline EosPayload/lib/mqtt/client.py.orig:45]: More merge conflicts that got left unresolved

File: EosPayload/drivers/RadioDriver.py [hunk 4]
```diff
@@ -1,7 +1,91 @@
...
+
+            # add packet to queue
+            priority = packet_from_mqtt.data_header.priority
+            logger.info(f"Enqueuing packet seq={self.sequence_number}")
+            self._thread_queue.put((priority, datetime.now(), packet_from_mqtt,))
+
+            self.sequence_number = (self.sequence_number + 1) % 256  # sequence number can't exceed 255
+
+        self._mqtt.register_subscriber(Topic.RADIO_TRANSMIT, xbee_send_callback)
+        self.port.add_data_received_callback(data_receive_callback)
```
Comment (thomasmholder) [inline EosPayload/drivers/RadioDriver.py:83]: If you move the above function definitions to outside of the device_read function, you can do all of this in setup, which is generally where MQTT stuff should be handled

File: EosPayload/drivers/PriorityQueue.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import itertools
```
Comment (thomasmholder) [inline EosPayload/drivers/PriorityQueue.py:1]: Why does this file exist

File: EosPayload/drivers/RadioDriver.py [hunk 6]
```diff
@@ -1,7 +1,91 @@
...
+    device_map = {
+        Device.RADIO: Topic.RADIO_TRANSMIT,
+        Device.MISC_RADIO_1: Topic.PING_COMMAND
+    }
+
+    def setup(self) -> None:
+        con = True
+        while con:
+            try:
+                self.port = XBeeDevice("/dev/cu.usbserial-8", 9600)
```
Comment (thomasmholder) [inline EosPayload/drivers/RadioDriver.py:34]: Need to update this to find the device

**Review by DarylDohner — CHANGES_REQUESTED** [review]

**Review by DarylDohner — DISMISSED** [review]

**Review by DarylDohner — APPROVED** [review]

**Pull request merged**